common/test/acceptance/tests/lms/test_lms_problems.py

```diff
@@ -86,3 +86,52 @@ def test_clarification(self):
         self.assertIn('Return on Investment', tooltip_text)
         self.assertIn('per year', tooltip_text)
         self.assertNotIn('strong', tooltip_text)
+
+
+class ProblemExtendedHintTest(ProblemsTest):
+    """
+    Test that extended hint features plumb through to the page html.
+    """
+    def get_problem(self):
+        """
+        Problem with extended hint features.
+        """
+        xml = dedent("""
+            <problem>
+            <p>question text</p>
+            <stringresponse answer="A">
+                <stringequalhint answer="B">hint</stringequalhint>
+                <textline size="20"/>
+            </stringresponse>
+            <demandhint>
+              <hint>demand-hint1</hint>
+              <hint>demand-hint2</hint>
+            </demandhint>
+            </problem>
+        """)
+        return XBlockFixtureDesc('problem', 'TITLE', data=xml)
+
+    def test_check_hint(self):
+        """
+        Test clicking Check shows the extended hint in the problem message.
+        """
+        self.courseware_page.visit()
+        problem_page = ProblemPage(self.browser)
+        self.assertEqual(problem_page.problem_text[0], u'question text')
+        problem_page.fill_answer('B')
+        problem_page.click_check()
+        self.assertEqual(problem_page.message_text, u'Incorrect: hint')
```

Comment:
Here's the excellent developer doc: https://edx.readthedocs.org/projects/edx-developer-guide/en/latest/analytics.html#bok-choy-testing